packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -487,16 +528,73 @@ export type TreeNodeFromImplicitAllowedTypes<
 
 /**
  * Type of content that can be inserted into the tree for a node of the given schema.
+ *
+ * @typeparam TSchemaInput - Schema to process.
+ * @typeparam T - Do not specify: default value used as implementation detail.
  * @public
  */
-export type InsertableTreeNodeFromImplicitAllowedTypes<
-	TSchema extends ImplicitAllowedTypes = TreeNodeSchema,
+export type InsertableTreeNodeFromImplicitAllowedTypesXXXX<
+	TSchemaInput extends ImplicitAllowedTypes,
+	TSchema = UnionToIntersection<TSchemaInput>,
 > = TSchema extends TreeNodeSchema
 	? InsertableTypedNode<TSchema>
 	: TSchema extends AllowedTypes
 		? InsertableTypedNode<FlexListToUnion<TSchema>>
 		: never;
 
+/**
+ * Marker for documenting when a type computed from a schema is contravariant
+ * with respect to the schema and thus only suitable for using for "input".
+ *
+ * Generic types which produce "Input" types error on the side of producing overly strict requirements.
+ * Future versions of the API can relax these requirements.
+ *
+ * For example if {@link InsertableField} produced `Input<never>` for some union of schema (ex: `A | OptionalField<A>`),
+ * a future version could instead return a more flexible but still safe type, like `Input<A>`.
+ *
+ * Note that since `Input` is just a genic type, it might not always appear in contexts which simplify or inline types.
+ * Thus "Input" in mainly useful as an annotation visible in type signatures and documentation.
+ *
+ * @privateRemarks
+ * TODO: Actually use this.
```

Comment:
I could use some advice on how to associate these docs with the relevant APIs.

Wrapping the output is a bit messy. I have `@see Input` docs, but with Input not being package exported the build is passing, which seems wrong. Maybe (unlike in intelisense) API-Extractor expects a link tag there?

Maybe an `Insertable` namespace with a doc comment would be viable, at least for the public ones?

Reply:
Left another comment before seeing this one. I'm surprised intellisense treats `@see` like a link - TSDoc does not, so you do need the `{@link}` tag.

Reply:
I like the namespace idea in theory, but I have a suspicion that using one will break API-Extractor in the `fluid-framework` package ☹️

https://github.com/microsoft/rushstack/issues/4807

Would it be problematic to just package-export this type as `@system` for no other reason than documentation?

Reply:
> Would it be problematic to just package-export this type as @system for no other reason than documentation?

No, I can do that. It seems a bit hacky to create linkable docs by exporting unused types just for their docs, but there's nothing inherently bad about it, and we can do something else layer since its a system type.

I guess I'll do that.